### ISSUE TITLE
[docs] Cannot generate MlirEnv class documentation because it is not defined

### ISSUE THREAD

**Issue by ChrisCummins** [facebookresearch/CompilerGym#688]
## 🐛 Bug

The `MlirEnv` class is only conditionally included in the compiler_gym package when built from source by CMake with the appropriate config option set. This means that `make docs` fails for MlirEnv.

## To Reproduce

1. `make docs`
2. Observer `AttributeError: MlirEnv` error:

```
WARNING: autodoc: failed to import class 'MlirEnv' from module 'compiler_gym.envs'; the following exception was raised:
Traceback (most recent call last):
  File "/private/home/cummins/.conda/envs/compiler_gym/lib/python3.8/site-packages/sphinx/util/inspect.py", line 412, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: module 'compiler_gym.envs' has no attribute 'MlirEnv'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/private/home/cummins/.conda/envs/compiler_gym/lib/python3.8/site-packages/sphinx/ext/autodoc/importer.py", line 110, in import_object
    obj = attrgetter(obj, mangled_name)
  File "/private/home/cummins/.conda/envs/compiler_gym/lib/python3.8/site-packages/sphinx/ext/autodoc/__init__.py", line 318, in get_attr
    return autodoc_attrgetter(self.env.app, obj, name, *defargs)
  File "/private/home/cummins/.conda/envs/compiler_gym/lib/python3.8/site-packages/sphinx/ext/autodoc/__init__.py", line 2606, in autodoc_attrgetter
    return safe_getattr(obj, name, *defargs)
  File "/private/home/cummins/.conda/envs/compiler_gym/lib/python3.8/site-packages/sphinx/util/inspect.py", line 428, in safe_getattr
    raise AttributeError(name) from exc
AttributeError: MlirEnv
```

## Environment

Please fill in this checklist:

- CompilerGym: v0.2.3
- How you installed CompilerGym (pip, source): source
- OS: Linux
- Python version: 3.8
